File: datamodel/src/main/java/datamodel/clinical/AtcClassification.java
package datamodel.clinical;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;

@Value.Immutable
public interface AtcClassification {

    @NotNull
    AtcLevel anatomicalMainGroup();

    @NotNull
    AtcLevel therapeuticSubGroup();

    @NotNull
    AtcLevel pharmacologicalSubGroup();

    @NotNull
    AtcLevel chemicalSubGroup();

    @NotNull
    AtcLevel chemicalSubstance();
}

File: algo/src/main/kotlin/com/hartwig/actin/datamodel/clinical/AtcClassification.kt
package com.hartwig.actin.datamodel.clinical

data class AtcClassification(
    val anatomicalMainGroup: AtcLevel,
    val therapeuticSubGroup: AtcLevel,
    val pharmacologicalSubGroup: AtcLevel,
    val chemicalSubGroup: AtcLevel,
    val chemicalSubstance: AtcLevel?
) {
    
    fun allLevels(): Set<AtcLevel> {
        return setOfNotNull(anatomicalMainGroup, therapeuticSubGroup, pharmacologicalSubGroup, chemicalSubGroup, chemicalSubstance)
    }
}
